js/src/views/Web/store.js

```diff
@@ -30,16 +31,20 @@ let instance = null;
 export default class Store {
   @observable counter = Date.now();
   @observable currentUrl = null;
+  @observable favorites = null;
   @observable history = [];
   @observable isLoading = false;
+  @observable isFavorited = false;
   @observable parsedUrl = null;
   @observable nextUrl = null;
   @observable token = null;
 
   constructor (api) {
     this._api = api;
 
+    this.favorites = localStore.get(FAVORITES_STORE) || {};
     this.nextUrl = this.currentUrl = this.loadLastUrl();
+    this.isFavorited = this.favorites[this.currentUrl];
```

Comment:
`isFavorited` should be computable, right now it's not updated when `this.currentUrl` changes (unless it's being changed by `setCurrentUrl`)